parser.cc

```diff
@@ -1453,6 +1454,52 @@ Input::Index AsciiFileInput::size() const
 	return file_size;
 }
 
+StreamInput StreamInput::Create(std::istream& s)
+{
+	const size_t start = s.tellg();
+	s.seekg(0, std::ios::end);
+
+	const size_t len = static_cast<size_t>(s.tellg()) - start;
+	s.seekg(start, std::ios::beg);
```

Comment:
Is start even non-zero?  Are we assuming that we can create an istream input starting at an arbitrary point in an input?